alacritty_terminal/src/tty/windows/mod.rs

```diff
@@ -33,161 +26,46 @@ use crate::tty::{ChildEvent, EventedPty, EventedReadWrite};
 
 mod child;
 mod conpty;
+#[cfg(feature = "winpty")]
+mod switchable_backends;
+#[cfg(feature = "winpty")]
 mod winpty;
 
-static IS_CONPTY: AtomicBool = AtomicBool::new(false);
+#[cfg(not(feature = "winpty"))]
+pub use conpty::Conpty as Backend;
+#[cfg(not(feature = "winpty"))]
+pub use mio_anonymous_pipes::{EventedAnonRead as ReadPipe, EventedAnonWrite as WritePipe};
 
-pub fn is_conpty() -> bool {
-    IS_CONPTY.load(Ordering::Relaxed)
-}
-
-enum PtyBackend {
-    Winpty(winpty::Agent),
-    Conpty(conpty::Conpty),
-}
+#[cfg(feature = "winpty")]
+pub use switchable_backends::{
+    EventedReadablePipe as ReadPipe, EventedWritablePipe as WritePipe, PtyBackend as Backend,
+};
```

Comment:
I'm not a fan of `pub use`'ing a bunch of renamed stuff here. It makes it incredibly difficult to see what types actually are and just sends people on a wild goose hunt to figure out what the types actually are.

Things should be re-imported as appropriate.

Reply:
Changed to just `use` instead of `pub use` for the renamed imports. They're now only used in this file, and I've added a private function `Pty::new` which takes args `impl Into<Backend>` etc. so that `conpty.rs` and `winpty.rs` don't need to import the concrete types.